core/vm/operations_acl.go

```diff
@@ -103,14 +104,26 @@ func makeGasSStoreFunc(clearingRefund uint64) gasFunc {
 func gasSLoadEIP2929(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (uint64, error) {
 	loc := stack.peek()
 	slot := common.Hash(loc.Bytes32())
+	var gasUsed uint64
+
+	if evm.chainConfig.IsCancun(evm.Context.BlockNumber) {
+		where := stack.Back(0)
+		addr := contract.Address()
+		index := trieUtils.GetTreeKeyStorageSlot(addr[:], where)
+		gasUsed += evm.Accesses.TouchAddressOnReadAndComputeGas(index)
+	}
+
 	// Check slot presence in the access list
 	if _, slotPresent := evm.StateDB.SlotInAccessList(contract.Address(), slot); !slotPresent {
 		// If the caller cannot afford the cost, this change will be rolled back
 		// If he does afford it, we can skip checking the same thing later on, during execution
 		evm.StateDB.AddSlotToAccessList(contract.Address(), slot)
-		return params.ColdSloadCostEIP2929, nil
+		gasUsed += params.ColdSloadCostEIP2929
+	} else {
+		gasUsed += params.WarmStorageReadCostEIP2929
```

Comment:
```suggestion
		return gasUsed + params.ColdSloadCostEIP2929, nil
	}
```

Reply:
in two parts because of a weird issue in the github interface. Anyhow, prefer returning early to `} else {`, that's another one of these go-ism that, I'm told, make it such a great language.